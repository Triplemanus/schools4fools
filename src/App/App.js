import React from 'react';
import './App.scss';
import { Route, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import SearchForm from '../Forms/SearchForm';
import SchoolCardContainer from '../Containers/SchoolCardContainer';
import { getAllSchools, isLoading, hasErrored } from '../actions';
import { fetchAllSchools } from '../apiCalls/apiCalls';

export class App extends React.Component {
  componentDidMount = async () => {
    try {
      const schools = await fetchAllSchools();
      console.log('fetched schools is: ', schools);
      getAllSchools(schools);
      isLoading(false);
    } catch ({ message }){
      isLoading(false);
      hasErrored(message);
    }
    }
    
    render() {
      return (
        <section className='App'>
        <Link to={"/"} className="header-link">
        <div className="headerContents">
              <h1>Schoolhouses Rock!</h1>
            </div>
          </Link> 
        <Route path="/" render={() => <SearchForm />} />
        <Link to={"/schools"}>
          <section>
            <p>All Matching Schools</p>
          </section>
        </Link>
        <Route path="/schools" render={() => <SchoolCardContainer />} />
      </section>
    )
  }
};

export const mapStateToProps = (state) => ({
  schools: state.schools,
  error: state.error
});

export const mapDispatchToProps = dispatch => ({
  isLoading: () => dispatch(isLoading()),
  getAllSchools: () => dispatch(getAllSchools()),
  hasErrored: () => dispatch(hasErrored())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);