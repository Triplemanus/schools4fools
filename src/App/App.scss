
html {
  height: 100%;
  overflow: hidden;
}

.App {
  text-align: center;
  /* background-: '../assets/Traditional brick schoolhouse.jpg' */
  position: relative;
  overflow: hidden;
}

.App img {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: auto;
  opacity: 0.6;
}

body {
  background-image: url('../assets/Traditional brick schoolhouse.jpg');
  /* background-color: rgb(80, 80, 245) ; */
  background-size: cover;
  background-attachment: fixed;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
}