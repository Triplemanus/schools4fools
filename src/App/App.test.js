import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';

describe('App', () => {
  let wrapper, mockSchool;

  beforeEach(() => {
   
    mockSchool = [
      {
        school_id: "080336006756",
        key: "080336006756",
        school_name: "George Washington",
        phone: '303.399.4532',
        address: '1234 S Monaco St.',
        distance: 5.8,
        low_Grade: '9',
        high_Grade: '12',
        level: 'High',
        is_Charter: false,
        is_Magnet: false,
        is_Private: false
      }
    ]

    wrapper = shallow(<App
      school={mockSchool}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });
});

describe ('mapStateToProps', () => {

  it('should return an object with the current school', () => {
    const mockState = {
      schools: [
        {
          school_id: "080336006756",
          key: "080336006756",
          school_name: "George Washington",
          phone: '303.399.4532',
          address: '1234 S Monaco St.',
          distance: 5.8,
          low_Grade: '9',
          high_Grade: '12',
          level: 'High',
          is_Charter: false,
          is_Magnet: false,
          is_Private: false
        }
      ],
      error: ''
    };

    const expectedSchools =  [{
        school_id: "080336006756",
        key: "080336006756",
        school_name: "George Washington",
        phone: '303.399.4532',
        address: '1234 S Monaco St.',
        distance: 5.8,
        low_Grade: '9',
        high_Grade: '12',
        level: 'High',
        is_Charter: false,
        is_Magnet: false,
        is_Private: false
      }];

      expect(mapStateToProps(mockState).schools).toEqual(expectedSchools);

  })
})

