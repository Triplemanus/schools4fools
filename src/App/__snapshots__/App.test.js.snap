// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<section
  className="App"
>
  <Link
    className="header-link"
    to="/"
  >
    <div
      className="headerContents"
    >
      <h1>
        Schoolhouses Rock!
      </h1>
    </div>
  </Link>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/schools"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/schools/:school_id"
    render={[Function]}
  />
</section>
`;
